src/modules/lua.rs

```diff
@@ -0,0 +1,115 @@
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::lua::LuaConfig;
+use crate::formatter::StringFormatter;
+use crate::utils;
+
+/// Creates a module with the current Lua version
+///
+/// Will display the Lua version if any of the following criteria are met:
+///     - Current directory contains a `.lua-version` file
+///     - Current directory contains a file with the `.lua` extension
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_lua_project = context
+        .try_begin_scan()?
+        .set_files(&[".lua-version"])
+        .set_extensions(&["lua"])
+        .is_match();
+
+    if !is_lua_project {
+        return None;
+    }
+
+    let mut module = context.new_module("lua");
+    let config = LuaConfig::try_load(module.config);
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    format_lua_version(&utils::exec_cmd("lua", &["-v"])?.stdout.as_str()).map(Ok)
+                }
+                _ => None,
+            })
+            .parse(None)
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `lua`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn format_lua_version(lua_stdout: &str) -> Option<String> {
+    // lua -v output looks like this:
+    // Lua 5.4.0  Copyright (C) 1994-2020 Lua.org, PUC-Rio
+
+    let version = lua_stdout
+        // split into ["", "5.4.0  Copyright (C) 1994-2020 Lua.org, PUC-Rio"]
+        .splitn(2, "Lua ")
```

Comment:
I'm not a lua dev, but is it possible for someone to have `luajit` as `lua` in their path? In that line of thinking a `lua_binary` option like in the `python module` might be useful.

This line wouldn't work with `luajit` because the version string is `LuaJIT 2.1.0-beta3 -- Copyright (C) 2005-2017 Mike Pall. http://luajit.org/` on my system

Reply:
Thanks for tips!
I'll check it out 👍